src/core/unit.c

```diff
@@ -3542,13 +3542,14 @@ static int get_name_owner_handler(sd_bus_message *message, void *userdata, sd_bu
 
         e = sd_bus_message_get_error(message);
         if (e) {
-                if (!sd_bus_error_has_name(e, SD_BUS_ERROR_NAME_HAS_NO_OWNER)) {
-                        r = sd_bus_error_get_errno(e);
+                r = sd_bus_error_get_errno(e);
+                if (sd_bus_error_has_names(e, SD_BUS_ERROR_NO_REPLY, SD_BUS_ERROR_DISCONNECTED))
+                        return log_unit_warning_errno(u, r, "GetNameOwner() failed, ignoring: %s",
+                                        bus_error_message(e, r));
```

Comment:
- Spurious line break ?
- Please add comments why we ignore the errors here.
- This is a callback. So please return 0 and do not propagate the negative errno.